src/schemas/text-based-fields.ts

```diff
@@ -1,3 +1,34 @@
+import {
+    EElementType,
+    IBaseTextBasedFieldAttributes,
+} from "src/context-providers";
+import { ELEMENT_VALIDATION_TYPES } from "src/data";
 import * as yup from "yup";
 
-export const EMAIL_SCHEMA = yup.object().shape({});
+const DOMAIN_REGEX = /^@[^\s]+(\.[^\s]+)*(\s*,\s*@([^\s]+(\.[^\s]+)*)+)*$/;
+
+export const EMAIL_SCHEMA = yup.object<IBaseTextBasedFieldAttributes>().shape({
+    placeholder: yup.string().optional(),
+    validation: yup.array().of(
+        yup.object().shape({
+            validationType: yup.string().required("Validation required."),
+
+            validationRule: yup.string().when("validationType", {
+                is: ELEMENT_VALIDATION_TYPES["Text field"][EElementType.EMAIL]
+                    .validationTypes[0],
+                then: (rule) =>
+                    rule
+                        .required("Email address required.")
+                        .matches(
+                            DOMAIN_REGEX,
+                            "Invalid email address format.  Check if email address is correct with no whitespace between characters. Separate each address with a comma if there is more than 1 email."
+                        ),
```

Comment:
the error messages are not accurate, they should be referring to the domains and not email addresses, may I trouble you to discuss with UX on this?

Reply:
have discussed will change it :)